app/uk/gov/hmrc/gform/models/FormatExpr.scala

```diff
@@ -84,3 +85,19 @@ object TextConstraint {
   implicit val format: OFormat[TextConstraint] = derived.oformat[TextConstraint]
 }
 
+object TextExpression {
+  implicit val format: OFormat[TextExpression] = {
+    val writes: OWrites[TextExpression] = derived.owrites
+    val stdReads = Json.reads[TextExpression]
+    val reads: Reads[TextExpression] = stdReads orElse Reads {
+      case JsString(expression) =>
+        BasicParsers.validateWithParser(expression, ValueParser.expr).fold(
```

Comment:
We are trying to remove parsing logic from case classes. Here in model we would like to have no logic at all since it is shared among other projects (gform-frontend, eeitt). Could you redesing it so parsing will go to `FieldValueMaker` leaving here only simple case class <-> Json formatter?